cmd/ftl/cmd_secret.go

```diff
@@ -169,3 +171,91 @@ func (s *secretUnsetCmd) Run(ctx context.Context, scmd *secretCmd, adminClient a
 	}
 	return nil
 }
+
+type secretImportCmd struct {
+	Input *string `arg:"" placeholder:"JSON" help:"JSON to import as secrets (read from stdin if omitted). Format: [{\"ref\":\"<module>.<name>\",\"value\": <secret>}, ...]" optional:""`
+}
+
+func (s *secretImportCmd) Help() string {
+	return `
+Imports secrets from a JSON array.
+`
+}
+
+func (s *secretImportCmd) Run(ctx context.Context, scmd *secretCmd, adminClient admin.Client) error {
+	var input []byte
+	var err error
+	if s.Input != nil {
+		input = []byte(*s.Input)
+	} else {
+		input, err = io.ReadAll(os.Stdin)
+		if err != nil {
+			return fmt.Errorf("failed to read config from stdin: %w", err)
+		}
+	}
+	var entries map[string]any
+	err = json.Unmarshal(input, &entries)
+	if err != nil {
+		return fmt.Errorf("could not parse JSON: %w", err)
+	}
+	for refPath, value := range entries {
+		ref, err := cf.ParseRef(refPath)
+		if err != nil {
+			return fmt.Errorf("could not parse ref %q: %w", refPath, err)
+		}
+		bytes, err := json.Marshal(value)
+		if err != nil {
+			return fmt.Errorf("could not marshal value for %q: %w", refPath, err)
+		}
+		req := &ftlv1.SetSecretRequest{
+			Ref:   configRefFromRef(ref),
+			Value: bytes,
+		}
+		if provider, ok := scmd.provider().Get(); ok {
+			req.Provider = &provider
+		}
+		_, err = adminClient.SecretSet(ctx, connect.NewRequest(req))
+		if err != nil {
+			return fmt.Errorf("could not import secret for %q: %w", refPath, err)
+		}
+	}
+	return nil
+}
+
+type secretExportCmd struct {
+}
+
+func (s *secretExportCmd) Help() string {
+	return `
+Outputs secrets in a JSON array. A provider can be used to filter which secrets are included.
+`
+}
+
+func (s *secretExportCmd) Run(ctx context.Context, scmd *secretCmd, adminClient admin.Client) error {
+	req := &ftlv1.ListSecretsRequest{
+		IncludeValues: optional.Some(true).Ptr(),
+	}
+	if provider, ok := scmd.provider().Get(); ok {
+		req.Provider = &provider
+	}
+	listResponse, err := adminClient.SecretsList(ctx, connect.NewRequest(req))
+	if err != nil {
+		return fmt.Errorf("could not retrieve secrets: %w", err)
+	}
+	entries := make(map[string]any, 0)
```

Comment:
Also not a JSON array as the docs suggest.

Reply:
Ah, yes the docs are out of date, will fix